File: 560. Subarray Sum Equals K/Java/Solution.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int sum = 0;
        int result = 0;
        Map<Integer, Integer> memory = new HashMap<>();
        memory.put(0, 1);
        for (int num : nums) {
            sum += num;
            if (memory.containsKey(sum - k)) {
                result += memory.get(sum - k);
            }
            if (!memory.containsKey(sum)) {
                memory.put(sum, 0);
            }
            memory.put(sum, memory.get(sum) + 1);
        }
        return result;
    }
}

File: 560. Subarray Sum Equals K/C++/Solution.cpp
class Solution {
public:
    int subarraySum(vector<int> &nums, int k) {
        int sum = 0;
        int result = 0;
        unordered_map<int, int> memory;
        memory[0] = 1;
        for (int num : nums) {
            sum += num;
            if (memory.count(sum - k)) {
                result += memory[sum - k];
            }
            if (!memory.count(sum)) {
                memory[sum] = 0;
            }
            memory[sum]++;
        }
        return result;
    }
};